### PR TITLE
feat: add no-new-symbol

### PR DESCRIPTION
**Pull request by makotot**
Add [no-new-symbol](https://eslint.org/docs/latest/rules/no-new-symbol) rule.

### CODE REVIEW

**Review by HerringtonDarkholme — COMMENTED** [review]
Thank you for adding a new rule! Appreciate it!
Would you like to add a line of `note` also?

File: rules/no-new-symbol.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+id: no-new-symbol
+message: Symbol cannot be called as a constructor.
+severity: error
+language: TypeScript
+rule:
+  pattern: new Symbol($$$)
```
Comment (HerringtonDarkholme) [inline rules/no-new-symbol.yml:6]: Can we also include the `note` field in the rule? Copy the text from eslint official site is good.

https://eslint.org/docs/latest/rules/no-new-symbol
```
Symbol is not intended to be used with the new operator, but to be called as a function.
```

Comment (makotot) [reply rules/no-new-symbol.yml:6]: 😮 I forgot to include it. Added it.

**Comment by makotot** [comment]
Thanks for the review! Sorry for the delay. I've made some corrections and would like you to take another look at it 🙏

**Review by HerringtonDarkholme — APPROVED** [review]
Thanks 👍

**Pull request merged**